sst/SimEngCoreWrapper.cc

```diff
@@ -255,7 +284,22 @@ void SimEngCoreWrapper::fabricateSimEngCore() {
     primaryComponentOKToEndSim();
     std::exit(EXIT_FAILURE);
   }
-
+// If testing is enabled populate heap if heap values have been specified.
+#ifdef SIMENG_ENABLE_SST_TESTS
+  if (heapStr_ != "") {
+    std::vector<uint8_t> initialHeapData;
+    std::vector<uint64_t> heapVals = splitHeapStr();
+    uint64_t heapSize = heapVals.size() * 8;
+    initialHeapData.resize(heapSize);
+    uint64_t* heap = reinterpret_cast<uint64_t*>(initialHeapData.data());
+    for (size_t x = 0; x < heapVals.size(); x++) {
+      heap[x] = heapVals[x];
+    }
```

Comment:
This mechanism is similar to what is done in all tests in `regression/aarch64/instructions`.
The need for casting arises as SST stores data in chunks of `uint8_t` bytes and initial heap data is of type `uint64_t`

**Quoting from [cppreference](https://en.cppreference.com/w/cpp/language/reinterpret_cast)**
> A value of any integral or enumeration type can be converted to a pointer type. A pointer converted to an integer of sufficient size and back to the same pointer type is guaranteed to have its original value, otherwise, the resulting pointer cannot be dereferenced safely (the round-trip conversion in the opposite direction is not guaranteed; the same pointer may have multiple integer representations) The null pointer constant [NULL](https://en.cppreference.com/w/cpp/types/NULL) or integer zero is not guaranteed to yield the null pointer value of the target type; [static_cast](https://en.cppreference.com/w/cpp/language/static_cast) or [implicit conversion](https://en.cppreference.com/w/cpp/language/implicit_cast) should be used for this purpose.

As far as the second point is concerned, heap and initialHeapData will always have the same pointer. After the cast is complete `uint64_t` values are then placed in  `std::vector<uint8_t>initialHeapData` using a casted `uint32_t*` start address.